via/services/jstor.py

```diff
@@ -36,10 +38,51 @@ def stream_pdf(self, url, jstor_ip):
             doi = f"{self.DEFAULT_DOI_PREFIX}/{doi}"
 
         url = f"{self._api_url}/{doi}?ip={jstor_ip}"
-        return self._http.stream(
+        stream = self._http.stream(
             url, headers=add_request_headers({"Accept": "application/pdf"})
         )
 
+        # Currently, JSTOR is sending us a stream of base 64 encoded data. This
+        # isn't intentional and may change if they can fix it. For the time
+        # being we need to decode this on the fly. This is both because it
+        # gives the user a better time to first byte, but also because the
+        # machinery that consumes this expects a stream, not a string.
+        return decode_base64_stream(stream)
+
+
+def decode_base64_stream(b64_stream):
```

Comment:
Nice, much better than reading the whole thing just to decode it (and change all the machinery around streaming for pdfs).

We might have to delete it if JSTOR changes back to honor the pdf header.